File: app/src/main/java/com/example/android/myaddressbook/Coctact_Loader.java
package com.example.android.myaddressbook;

import android.content.AsyncTaskLoader;
import android.content.Context;

import java.io.IOException;
import java.io.InputStream;

/**
 * Created by Shami on 10/7/2017.
 */

public class Coctact_Loader extends AsyncTaskLoader<String> {

    Context mContext;

    public Coctact_Loader(Context context) {
        super(context);
        mContext=context;
    }

    @Override
    protected void onStartLoading() {
        super.onStartLoading();

        forceLoad();
    }

    @Override
    public String loadInBackground() {

        String response=readContactJsonFile();
        return response;
    }

    public  String  readContactJsonFile() {
        String contactsString = null;
        try {
            InputStream inputStream = mContext.getAssets().open("mock_contacts.json");
            int size = inputStream.available();
            byte[] buffer = new byte[size];
            inputStream.read(buffer);
            inputStream.close();

            contactsString = new String(buffer);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return contactsString;
    }


}

File: app/src/main/java/com/example/android/myaddressbook/Coctact_Loader.kt
package com.example.android.myaddressbook

import android.content.AsyncTaskLoader
import android.content.Context
import java.io.IOException

/**
 * Created by Shami on 10/7/2017.
 */

class Coctact_Loader(internal var mContext: Context) : AsyncTaskLoader<String>(mContext) {

    override fun onStartLoading() {
        super.onStartLoading()

        forceLoad()
    }

    override fun loadInBackground(): String? {

        return readContactJsonFile()
    }

    fun readContactJsonFile(): String? {
        var contactsString: String? = null
        try {
            val inputStream = mContext.assets.open("mock_contacts.json")
            val size = inputStream.available()
            val buffer = ByteArray(size)
            inputStream.read(buffer)
            inputStream.close()

            contactsString = String(buffer)
        } catch (e: IOException) {
            e.printStackTrace()
        }

        return contactsString
    }


}
